prog/weaver/main.go

```diff
@@ -246,6 +253,19 @@ func main() {
 	}
 	config.ProtocolMinVersion = byte(protocolMinVersion)
 
+	if !proxyConfig.Disabled {
+		// Start Weave Proxy:
+		proxy, err := weaveproxy.NewProxy(*proxyConfig)
+		if err != nil {
+			Log.Fatalf("Could not start proxy: %s", err)
+		}
+		defer proxy.Stop()
+		listeners := proxy.Listen()
```

Comment:
That's what I wanted to do as a subsequent refactoring. Now may be the right time given it seems to be in a functional state.